packages/react-magma-dom/src/components/Accordion/useAccordionButton.ts

```diff
@@ -0,0 +1,131 @@
+import { isArray } from 'lodash';
+import * as React from 'react';
+
+import { AccordionContext } from './useAccordion';
+import { AccordionItemContext } from './useAccordionItem';
+import { useForceUpdate, useForkedRef } from '../../utils';
+
+export interface UseAccordionButtonProps {
+  testId?: string;
+  isInverse?: boolean;
+}
+
+export function useAccordionButton(
+  props: UseAccordionButtonProps,
+  forwardedRef
+) {
+  const {
+    buttonRefArray,
+    expandedIndex,
+    iconPosition,
+    isControlled,
+    isMulti,
+    registerAccordionButton,
+    onExpandedChange,
+    setExpandedIndex,
+  } = React.useContext(AccordionContext);
+
+  const { buttonId, index, isDisabled, isExpanded, panelId } = React.useContext(
+    AccordionItemContext
+  );
+
+  const handleClick = () => {
+    if (typeof onExpandedChange === 'function') {
```

Comment:
can we use this the same way we do elsewhere?  `x && typeof x === 'function' && x()`

Reply:
We sure can!